Car.java

```diff
@@ -0,0 +1,94 @@
+public class Car {
+
+    private String color;
+    private String mark;
+    private int consumption;
+    private double volume;
+    private double mileage;
+
+
+    public Car(String color, String mark, int cons, double vol, double mil) {
+        this.color = color;
+        this.mark = mark;
+        this.consumption = cons;
+        this.volume = vol;
+        this.mileage = mil;
+    }
+
+
+    public void drive(int km) {
+
+        double volumeForTheTrip = consumption * km;
+        double fuelLeft = volume - volumeForTheTrip;
+
+
+        if (fuelLeft > 0) {
+            System.out.println("Проехала: " + km + " км\n");
+            System.out.println("Пробег: " + (km + mileage) + "км\n");
+            mileage = mileage + km;
+            volume = volume - volumeForTheTrip;
+        } else if (fuelLeft <= 0) {
+            double distance = volume / consumption;
+            System.out.println("Проехала: " + distance + " км\n");
+            System.out.println("Пробег: " + (distance + mileage) + "км\n");
+            mileage = mileage + distance;
+            volume = 0;
+        } else if (consumption > volume || volume < volumeForTheTrip || volume == 0) {

```

Comment:
volumeForTheTrip = consumption * km;
fuelLeft = volume - volumeForTheTrip;

If _(consumption > volume)_ => _(volumeForTheTrip  > volume)_ automatically (or non-positive value is set to km) .
Then **fuelLeft** will be negative and _else if (fuelLeft <= 0)_